client/src/app/pages/applications/components/application-form/application-form.tsx

```diff
@@ -445,78 +447,36 @@ export const ApplicationForm: React.FC<ApplicationFormProps> = ({
-                  what: t("terms.tags").toLowerCase(),
-                })}
-                searchInputAriaLabel="tags-select-toggle"
-                selections={
-                  value
-                    .map(
-                      (formTag) =>
-                        tags?.find((tagRef) => tagRef.name === formTag.name)
-                    )
-                    .map((matchingTag) =>
-                      matchingTag ? matchingTag.name : undefined
-                    )
-                    .filter((e) => e !== undefined) as string[]
-                }
-              />
-              // <SimpleSelect
-              //   maxHeight={DEFAULT_SELECT_MAX_HEIGHT}
-              //   placeholderText={t("composed.selectMany", {
-              //     what: t("terms.tags").toLowerCase(),
-              //   })}
-              //   id="tags-select"
-              //   variant="typeaheadmulti"
-              //   toggleId="tags-select-toggle"
-              //   toggleAriaLabel="tags dropdown toggle"
-              //   aria-label={name}
-              //   value={value
-              //     .map((formTag) =>
-              //       tags?.find((tagRef) => tagRef.name === formTag.name)
-              //     )
-              //     .map((matchingTag) =>
-              //       matchingTag
-              //         ? {
-              //             value: matchingTag.name,
-              //             toString: () => matchingTag.name,
-              //           }
-              //         : undefined
-              //     )
-              //     .filter((e) => e !== undefined)}
-              //   options={tagOptions}
-              // onChange={(selection) => {
-              //   const selectionWithValue = selection.toString();
-
-              //   const currentValue = value || [];
-              //   const e = currentValue.find(
-              //     (f) => f.name === selectionWithValue
-              //   );
-              //   if (e) {
-              //     onChange(
-              //       currentValue.filter((f) => f.name !== selectionWithValue)
-              //     );
-              //   } else {
-              //     const tag = getTagRef(selectionWithValue);
-              //     if (currentValue && typeof tag !== "undefined")
-              //       onChange([...currentValue, tag]);
-              //   }
-              // }}
-              //   onClear={() => onChange([])}
-              //   noResultsFoundText={t("message.noResultsFoundTitle")}
-              // />
-            )}
+            renderInput={({ field: { value, name, onChange } }) => {
+              const selections = value
+                .map(
+                  (formTag) =>
+                    tags?.find((tagRef) => tagRef.name === formTag.name)
+                )
+                .map((matchingTag) =>
+                  matchingTag ? matchingTag.name : undefined
+                )
+                .filter((e) => e !== undefined) as string[];
```

Comment:
```suggestion
              const tagNames = new Set(tags?.map((tag) => tag.name));

              const selections = value.flatMap((formTag) =>
                formTag.source === "" && tagNames.has(formTag.name)
                  ? [formTag.name]
                  : []
              );

```

Reply:
Played around with this a bit and borrowed Mikes suggestion of using a Set/has here instead of creating a map and using includes since set narrows the tags to unique values & boosts performance. Explicitly checking for the empty string on the form tag on lookup should give us what we want since empty string here means "manually selected tag" for a tag on an application object (a formTag in this case) .

Reply:
I ended up making `tagOptions` a set of strings (tag names) from the beginning and using that. Also as the onChange event maps the form values to `tags` (which seems to have undefined source instead of ""), I merge a `{source:""}` object when returning from `getTagRef`. Without this, after making any changes to the value we'd filter them out if they didn't have a source prop in the `tags` array.

Reply:
Noticing that onSubmit, any tags that are filtered out from the selections on line 448 are not being added back in to the payload.

Reply:
Sam confirmed that there is no way for us to overwrite those non-manual tags by updating the application object directly so we should be good on what we are passing to the put request here. There seems to be some ongoing discussion/weirdness around this since the non-manual tags are "virtual" tags being inherited from other membership criteria etc. 



We do need to make sure we are filtering out any pre-existing, non-manual tags in the selectable tag list so that we can't accidentally create a new "manual" tag for a tag that already exists on the application from a different source.

Reply:
I'm unsure if there is a change you're asking for or if you are saying to wait.

Right now it is using `tags` (filtered to only include undefined or "" source value) as the available options and not letting the user add new tags that don't exist outside the options.

Reply:
Understood - My concern was that after we are filtering out the tags on an app with a source other than "", we never add them back in to the tags payload on submit so we'd be overwriting those tags.

Reply:
>Right now it is using tags (filtered to only include undefined or "" source value) as the available options and not letting the user add new tags that don't exist outside the options.

The change I think we might temporarily need is to check to see if any tags in the "tagOptions" list you've created match up with any other tags on the application (archetype source or other non-manual tags) and remove any matching available tag from the list. Modifying the tags api fetch response won't include any tags with other sources . Those sources are only on tags that have been associated with an application or archetype. We would have to cross check the response from the tags fetch api with what exists on the application in this case.